pkg/server/local_observer.go

```diff
@@ -394,7 +395,7 @@ func (r *flowsReader) Next(ctx context.Context) (*pb.Flow, error) {
 
 // newRingReader creates a new RingReader that starts at the correct ring
 // offset to match the flow request.
-func newRingReader(ring *container.Ring, req *observer.GetFlowsRequest) (*container.RingReader, error) {
+func newRingReader(ctx context.Context, ring *container.Ring, req *observer.GetFlowsRequest) (*container.RingReader, error) {
```

Comment:
I'm not fond of the idea of passing a context to a constructor. Why does `BuildFilter` require a context in the first place? I mean, I noticed that `OnBuildFilter` requires a context but I don't understand why. Is building filters an operation that is expected to be running for a very long time and that needs to be cancelable?

Reply:
I don't really understand the constructor comment. The GetFlows function handles a request and the context is passed as the request is being handled. `newRingReader()` is part of this, it only exists for the duration of the API request and the context is required in the `OnBuildFilter()`.

Reply:
Fixed as discussed on Slack